zerver/decorator.py

```diff
@@ -417,12 +418,12 @@ def _wrapped_func_arguments(request, *args, **kwargs):
                 return json_unauthorized(e.error)
             request.user = profile
             process_client(request, profile)
-            if isinstance(profile, UserProfile):
-                request._email = profile.email
-            else:
+            if is_remote_server(role):
                 assert isinstance(profile, RemoteZulipServer)  # type: ignore # https://github.com/python/mypy/issues/2957
                 request._email = "zulip-server:" + role
                 profile.rate_limits = ""
+            else:
+                request._email = profile.email
```

Comment:
I'm not totally on board with this change, since it loses the typing check we got from `isinstance`.